sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/aad/AADClientRegistrationRepository.java

```diff
@@ -199,21 +199,31 @@ public static int resourceServerCount(Set<String> scopes) {
                                      .orElseGet(Stream::empty)
                                      .collect(Collectors.toSet());
         result.addAll(azureClientOpenidScopes());
-        if (properties.allowedGroupIdsConfigured() || properties.allowedGroupNamesConfigured()) {
-            // The 2 scopes are need to get group name from graph.
-            result.add(properties.getGraphBaseUri() + "User.Read");
+        // About "Directory.Read.All" and "User.Read", please refer to:
+        // 1. https://docs.microsoft.com/en-us/graph/permissions-reference
+        // 2. https://github.com/Azure/azure-sdk-for-java/issues/21284#issuecomment-888725241
+        if (properties.allowedGroupNamesConfigured()) {
+            // "Directory.Read.All" allows to get group id and group name.
             result.add(properties.getGraphBaseUri() + "Directory.Read.All");
+        } else if (properties.allowedGroupIdsConfigured()) {
```

Comment:
Is it possible both scopes need?

Reply:
No. `User.Read` is not necessary anymore we have `Directory.Read.All`.

Please refer to the link in line 204.

Reply:
@moarychan , could you  please comment `LGTM` if you think this PR is OK to merge?